### ISSUE TITLE
Failed to execute 'removeChild'

### ISSUE THREAD

**Issue by MarcGodard** [donejs/portal-ssr#5]
My `index.stache` has the following head:

```
  <head>
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, maximum-scale=1.0" />
    <meta charset="utf-8" />
    {{#if localesAvailable.length}}
      {{#each (localesAvailable)}}
        {{^eq scope.top.currentLocale code}}
          <link rel="alternate" hreflang="{{code}}" href="http://{{code}}.{{scope.top.domain}}{{scope.top.currentPage}}" />
        {{/eq}}
      {{/each}}
    {{/if}}
    <link rel="canonical" href="{{siteUrl}}{{currentPage}}" />
  </head>
```

When I switch language this head needs to refresh, however, I get the following error:

```
Uncaught DOMException: Failed to execute 'removeChild' on 'Node': The node to be removed is not a child of this node.
    at HTMLHeadElement.normal.(anonymous function) (http://localhost:8080/node_modules/can-dom-mutate/node/node.js:79:27)
    at Object.remove (http://localhost:8080/node_modules/can-view-nodelist/can-view-nodelist.js:491:26)
    at Object.replace (http://localhost:8080/node_modules/can-view-nodelist/can-view-nodelist.js:470:14)
    at updateNodeList (http://localhost:8080/node_modules/can-view-live/lib/html.js:26:13)
    at CompletionQueue.flush (http://localhost:8080/node_modules/can-queues/completion-queue.js:26:12)
    at Object.onComplete (http://localhost:8080/node_modules/can-queues/can-queues.js:46:16)
    at PriorityQueue.flush (http://localhost:8080/node_modules/can-queues/priority-queue.js:126:19)
    at Object.onComplete (http://localhost:8080/node_modules/can-queues/can-queues.js:32:16)
    at Queue.flush (http://localhost:8080/node_modules/can-queues/queue.js:62:17)
    at Object.stop (http://localhost:8080/node_modules/can-queues/can-queues.js:94:19)
```

I would prefer not needing to put this in every page, unless I have too. When I remove this `{{#if...}}` block, everything works as expected. I will look into an alternative to this. However, I thought I would report this.

**Comment by matthewp** [comment]
Thanks, this is likely a bug in can-stache, but i'm not sure exactly. Which variable are you changing? Is it `scope.top.currentLocale`?

**Comment by MarcGodard** [comment]
yes. Changing `scope.top.currentLocale` refreshes the list of `<link rel="alternate"...` in the head, but the head belongs to the portal now maybe? If I do this in the `{{portal}}` tags (the child) it works fine. So I made a component for it and adding it to all the child page's portals instead of the main stache. I think adding an error like "Using portal doesn't allow dynamic content in the main head" or something would be fine if its not an easy fix.

**Comment by matthewp** [comment]
The head doesn't belong to the portal, or at least shouldn't. It seems like there's some conflict here but i'll have to investigate what that is.

**Comment by matthewp** [comment]
What triggers `scope.top.currentLocale` to change? Is it a user interaction, like changing page? is it that you load something that triggers this property change?

**Comment by matthewp** [comment]
I can recreate now.

**Comment by matthewp** [comment]
This will be fixed in the next canjs version.

**Comment by matthewp** [comment]
This was a separate bug unrelated to portals. It was fixed in can `5.21.3`.
